src/main.ts

```diff
@@ -278,26 +296,25 @@ export class ConnectionFeature extends SqlOpsFeature<undefined> {
 			return Promise.resolve();
 		};
 
-		let registerOnConnectionComplete = (handler: (connSummary: azdata.ConnectionInfoSummary) => any): void => {
-			client.onNotification(protocol.ConnectionCompleteNotification.type, handler);
+		const onConnectionCompleteEventEmitter = this.registerNotificationEmitter(protocol.ConnectionCompleteNotification.type);
+		let registerOnConnectionComplete = (handler: (connSummary: azdata.ConnectionInfoSummary) => any): vscode.Disposable => {
+			return onConnectionCompleteEventEmitter.event(handler);
 		};
 
-		let registerOnIntelliSenseCacheComplete = (handler: (connectionUri: string) => any): void => {
-			client.onNotification(protocol.IntelliSenseReadyNotification.type, (params: types.IntelliSenseReadyParams) => {
-				handler(params.ownerUri);
-			});
+		const onIntelliSenseReadyEventEmitter = this.registerNotificationEmitter(protocol.IntelliSenseReadyNotification.type);
+		let registerOnIntelliSenseCacheComplete = (handler: (connectionUri: string) => any): vscode.Disposable => {
```

Comment:
Can these return `vscode.Event` type instead of 'any'?

Reply:
This is the return value of the handler itself. Fixing this would require mostly changes on the ADS side to fix the typings there and then port them over here (which would be great to do, just out of the scope of this PR)